lib/index.js

```diff
@@ -130,4 +133,87 @@ Dynamoose.prototype.revertDDB = function () {
   this.dynamoDB = null;
 };
 
+Dynamoose.prototype.transaction = async function(items, options, next) {
+  debugTransaction('Run Transaction');
+  const deferred = Q.defer();
+
+  options = options || {};
+  if(typeof options === 'function') {
+    next = options;
+    options = {};
+  }
+
+  if(items === null || items === undefined || items.length === 0) {
+    deferred.reject(new errors.TransactionError('Items required to run transaction'));
+    return deferred.promise.nodeify(next);
+  }
+
+  items = await Promise.all(items);
+  let transactionReq = {
+    TransactItems: items.map(item => {
+      const returnItem = {...item};
+      delete returnItem.$__;
+      return returnItem;
+    })
+  };
+  let transactionMethodName = items.map(obj => Object.keys(obj)[0]).every(key => key === "Get") ? "transactGetItems" : "transactWriteItems";
```

Comment:
Also - we should probably only do this computation _if_ `options.type` is not specified. i.e:
```js
let transactionMethodName;

if (options.type) {
  debugTransaction("Using custom transaction method");
  // ...
} else {
  debugTransaction("Using predetermined transaction method");
  transactionMethodName = // ...
}
```

Reply:
@jutaz I don't think so actually. We want the default to be `items.map(obj => Object.keys(obj)[0]).every(key => key === "Get") ? "transactGetItems" : "transactWriteItems"` I believe. Then if `options.type` is specified, we override that. Otherwise we could end up where `transactionMethodName` is null or undefined.

Reply:
@jutaz Just to make sure I'm clear. The major benefit to changing this is performance, correct?

Did address in latest commit.

Reply:
> Just to make sure I'm clear. The major benefit to changing this is performance, correct?

Performance, and also - this makes reading the code much easier: you have 2 logical branches that the code can possibly go, thus making it clearer to understand, based on external factors (in this case `options.type`'s value) which branch is executing.

Reply:
@jutaz Got it!